node_modules/oae-core/editprofile/js/editprofile.js

```diff
@@ -101,8 +101,12 @@ define(['jquery', 'oae.core'], function ($, oae) {
         var renderEditProfileMainPanel = function() {
             // If the display name is not valid, clear it to inform the template that the user
             // has no real display name
+            var tenantVisibility = oae.api.config.getValue('oae-principals', 'user', 'visibility');
             if (!isValidDisplayName()) {
                 profile.displayName = null;
+                if(profile.visibility === 'private' && tenantVisibility !== 'private') {
```

Comment:
But it might actually be easier to just always do `profile.visibility = tenantVisibility;` when the display name is not valid.